huaweicloud/services/workspace/resource_huaweicloud_workspace_desktop_name_rule.go

```diff
@@ -0,0 +1,238 @@
+package workspace
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+// @API Workspace POST /v2/{project_id}/desktop-name-policies
+// @API Workspace GET /v2/{project_id}/desktop-name-policies
+// @API Workspace PUT /v2/{project_id}/desktop-name-policies/{policy_id}
+// @API Workspace POST /v2/{project_id}/desktop-name-policies/batch-delete
+func ResourceDesktopNameRule() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceDesktopNameRuleCreate,
+		ReadContext:   resourceDesktopNameRuleRead,
+		UpdateContext: resourceDesktopNameRuleUpdate,
+		DeleteContext: resourceDesktopNameRuleDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name_prefix": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"digit_number": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"start_number": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"single_domain_user_increment": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"is_default_policy": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Computed: true,
+			},
+			"is_contain_user": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceDesktopNameRuleCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg     = meta.(*config.Config)
+		region  = cfg.GetRegion(d)
+		httpUrl = "v2/{project_id}/desktop-name-policies"
+	)
+
+	client, err := cfg.WorkspaceV2Client(region)
+	if err != nil {
+		return diag.Errorf("error creating Workspace v2 client: %s", err)
+	}
```

Comment:
```suggestion
	client, err := cfg.NewServiceClient(region)
	if err != nil {
		return diag.Errorf("error creating Workspace client: %s", err)
	}
```
The version info is never used in the createPath building phases

Reply:
fixed